src/sentry/snuba/outcomes.py

```diff
@@ -267,19 +272,39 @@ def __init__(
 
 
 def run_outcomes_query_totals(query: QueryDefinition) -> ResultSet:
```

Comment:
let's go ahead and make the timeseries query use the new query API as well -- easier to maintain going forward. It would also be ideal if the new logic could be moved into the QueryDefinition, and all this function would do is run the query. -- this may mean you need to change the interface of QueryDefinition -- that's fine.

Reply:
for sure, i can do that